src/map/script.cpp

```diff
@@ -21473,8 +21473,13 @@ BUILDIN_FUNC(instance_npcname)
 
 	if( instance_id > 0 && (nd = npc_name2id(str)) != NULL ) {
 		static char npcname[NAME_LENGTH];
-		snprintf(npcname, sizeof(npcname), "dup_%d_%d", instance_id, nd->bl.id);
-		script_pushconststr(st,npcname);
+		if(nd->instance_id > 0){
+			strncpy(npcname,str,NAME_LENGTH);
+			ShowWarning("buildin_instance_npcname: npc is already in instance and its name has a duplicate format, returning input name (instance_id: %d, NPC name: \"%s\".)\n", instance_id, str);
```

Comment:
I would rather be for SCRIPT_CMD_FAILURE and telling users to learn how to use the script command.
But this script command could be improved in general x_x